packages/@aws-cdk/aws-lambda-nodejs/README.md

```diff
@@ -0,0 +1,56 @@
+## Amazon Lambda Node.js Library
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> **This is a _developer preview_ (public beta) module. Releases might lack important features and might have
+> future breaking changes.**
+>
+> This API is still under active development and subject to non-backward
+> compatible changes or removal in any future version. Use of the API is not recommended in production
+> environments. Experimental APIs are not subject to the Semantic Versioning model.
+
+---
+<!--END STABILITY BANNER-->
+
+This library provides constructs for Node.js Lambda functions.
+
+### Node.js Function
+Define a `NodejsFunction`:
+
+```ts
+new lambda.NodejsFunction(this, 'my-handler');
+```
+
+By default, the construct will look up for a `index.ts` or `index.js` that exports a `handler`
+function in the folder named as the construct's id:
```

Comment:
I was actually thinking (albeit don't know for sure that this is the right approach) that the default behavior should be to use the name of the defining file with some suffix. You will need to roll back the stack trace to find the name of the defining file name but that should be possible.

```
.
├── boom.ts             # this is where NodejsFunction is defined
├── boom.lambda.ts .    # this is where `handler` is exported
```

Since we now have tree shaking and all that nice stuff, I'd argue that creating a special directory for each lambda function is a big hassle. Thoughts?

Reply:
In your example, `lambda` is the construct's id of the `NodejsFunction` in `boom.ts`?

Reply:
No, the name of the lambda handler file is based on the name of the declaring file. Requires a bit of stack trace magic.

Reply:
And if there are multiple `NodejsFunction` declared in `boom.ts`?

Reply:
Well....  didn’t think that far out :-)

I guess using the construct id as the suffix might be a good solution.

Reply:
Used the construct id, see latest changes